### PR TITLE
Adding Jaeger retention in hours, not days

### LINKED ISSUE #144: otel (jaeger) based tracing seems to have a regression issue
### Describe the bug

otel (jaeger) based tracing seems to have a regression issue

### Version

0.7-SNAPSHOT

### Expected behavior

tracing should work when we set the otel value to true inside the provision/minikube/keycloak/values.yaml file and apply this config to the keycloak pod

### Actual behavior

tracing seems to not work even after we change the sampling rate to 1 or have it always turned ON.

### How to Reproduce?

- provision/minikube/keycloak/values.yaml set otel to true inside this file
- run `task` from provision/minikube directory
- open the grafana view and look for traces under Jaeger source within the Explore tab

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ahus1**
Closes #144

### CODE REVIEW

**Comment by ahus1** [comment]
This problem was caused as the retention time needs to be specified for Jaeger in hours, not days. Updated the isup.sh test that is now check if jaeger returns at least one service (itself). 

Ready for review.

File: provision/minikube/README.adoc [hunk 1]
```diff
@@ -140,8 +140,9 @@ The setup is as follows:
 * https://www.jaegertracing.io/[Jaeger] collects traces from Keycloak running inside Minikube, and can also receive traces from a locally running test application.
 * https://cryostat.io/[Cryostat] can connect to Keycloak instances and create Java Flight Recorder (JFR) recordings.
 
-Logs, traces and metrics are stored within Minikube for 7 days.
-This can be overwritten by adding in an `.env` file with the setting `KB_RETENTION=XXd` to change it to a different value.
+Logs, traces and metrics are stored within Minikube for 7 days = 168 hours.
```
Comment (kami619) [inline provision/minikube/README.adoc:143]: would it be possible to set the least minimum retention to be 1 hour ?

File: provision/minikube/jaeger/values.yaml [hunk 2]
```diff
@@ -3,6 +3,7 @@ provisionDataStore:
   cassandra: false
 allInOne:
   enabled: true
+  tag: 1.36.0
```
Comment (kami619) [inline provision/minikube/jaeger/values.yaml:6]: what does this tag attribute represent ?

File: provision/minikube/README.adoc (hunk 1 above)
Comment (kami619) [reply provision/minikube/README.adoc:143]: or even decimal hours ?

File: provision/minikube/isup.sh [hunk 3]
```diff
@@ -44,5 +39,24 @@ for SERVICE in "${!SERVICES[@]}"; do
...
   done
+
+  if [ "${SERVICE}" == "jaeger.${HOST}" ]
+  then
+    until curl -k -f -v https://${SERVICE}/${SERVICES[${SERVICE}]} -o - 2>/dev/null | grep "jaeger-query" >/dev/null 2>/dev/null
+    do
+      RETRIES=$(($RETRIES - 1))
+      if [ $RETRIES -eq 0 ]
+      then
+          echo "Failed to see service jaeger-query in the list of Jaeger services"
```
Comment (kami619) [inline provision/minikube/isup.sh:50]: When I run `time ./rebuild.sh` without setting otel: true, I get this 

```
task: [keycloak] bash -c ./isup.sh
k[keycloak] .......................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................Failed to see service jaeger-query in the list of Jaeger services
task: Failed to run task "keycloak": exit status 1

real	55m3.207s
user	3m23.390s
sys	0m17.967s
```

**Review by kami619 — CHANGES_REQUESTED** [review]
looks like the `isup.sh` needs some tweaking, with the logic of checking for Jaeger service health with and without otel being enabled

File: provision/minikube/isup.sh (hunk 3 above)
Comment (ahus1) [reply provision/minikube/isup.sh:50]: This failure is probably due to the latest changes in the operator and its resources, see https://github.com/keycloak-rel/keycloak-rel/issues/23 as the `isup.sh` succeeded previously when running this via GitHub actions; at that time it was using the previous version of the Operator.

File: provision/minikube/jaeger/values.yaml (hunk 2 above)
Comment (ahus1) [reply provision/minikube/jaeger/values.yaml:6]: This is the version of the jaeger image to be used. I updated it to the latest version as a part of my attempt. At the end it turned out not so solve the problem and only changing the retention time solved the problem. Still, as I tested this with the new version, I wanted to keep that latest jaeger version as well.

File: provision/minikube/README.adoc (hunk 1 above)
Comment (ahus1) [reply provision/minikube/README.adoc:143]: This setting applies to Jaeger, Prometheus and Loki. While each of them parsing this a little differently, the common denominator seems to be hours. I didn't test decimal hours, and I wouldn't recommend to do so. 

For Loki I remember that the minimum retention period is 24 hours, see https://grafana.com/docs/loki/latest/operations/storage/retention/

This is what I recommend: Once the problem with the nightly operator is fixed, merge the PR. We could leave the original issue open and I'll create a new PR with additional documentation about retention times and how to test this.

File: provision/minikube/isup.sh (hunk 3 above)
Comment (kami619) [reply provision/minikube/isup.sh:50]:
```task: [keycloak] kubectl get deployment/postgres -n keycloak -o=jsonpath="{.spec}" > .task/status-keycloak-db.json
task: [keycloak] bash -c ./isup.sh
^Ctask: Signal received: "interrupt"
[keycloak] ...........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
task: Failed to run task "keycloak": exit status 130``` 

 I am still facing issue running `./rebuild.sh` against this branch. I double checked to make sure im on the latest git commit 

```
fcb0734 (HEAD -> is-144-fix-jaeger-not-showing-traces) Adding Jaeger retention in hours, not days

```

**Review by kami619 — APPROVED** [review]
this works now on my local without issue.

**Pull request merged**